### ISSUE TITLE
How to get object using idempotent key?

### ISSUE THREAD

**Issue by Kingg22** [oracle/oci-dotnet-sdk#276]
```cs
using var client = new ObjectStorageClient(Provider, new());
var getObjectRequest = new GetObjectRequest()
{
    BucketName = BucketName,
    NamespaceName = Namespace,
    ObjectName = objectName,
};

try
{
    var response = await client.GetObject(getObjectRequest);
}
```

Hi, I'm making a request like this, but I'd like to get an object with an ETag or a custom key with a more idempotent id to make sure I delete it correctly and don't lose other objects. Can this be done at the moment or could it be implemented?

**Comment by github-anurag** [comment]
@Kingg22 
You can use the [IfMatch](https://github.com/oracle/oci-dotnet-sdk/blob/master/Objectstorage/requests/GetObjectRequest.cs#L67) property of the `GetObjectRequest` to pass in the ETag as desired.

Please refer to the official [API documentation](https://docs.oracle.com/en-us/iaas/api/#/en/objectstorage/20160918/Object/GetObject) for more information.

**Comment by Kingg22** [comment]
> @Kingg22 
> You can use the [IfMatch](https://github.com/oracle/oci-dotnet-sdk/blob/master/Objectstorage/requests/GetObjectRequest.cs#L67) property of the `GetObjectRequest` to pass in the ETag as desired.
> 
> Please refer to the official [API documentation](https://docs.oracle.com/en-us/iaas/api/#/en/objectstorage/20160918/Object/GetObject) for more information.
> 
> 

Yes, I saw that attribute, but my question is if you could use it to search without using the name. Just the ID.

**Comment by github-anurag** [comment]
@Kingg22 
The `name` property is a required param for the `GetObjectRequest ` request

For your usecase perhaps you can try using [ListObjects](https://docs.oracle.com/en-us/iaas/api/#/en/objectstorage/20160918/Object/ListObjects) and pass in the `ETag` in the `fields` parameter

**Comment by Kingg22** [comment]
> @Kingg22 The `name` property is a required param for the `GetObjectRequest ` request
> 
> For your usecase perhaps you can try using [ListObjects](https://docs.oracle.com/en-us/iaas/api/#/en/objectstorage/20160918/Object/ListObjects) and pass in the `ETag` in the `fields` parameter

Thanks for that, anyway I wanted to leave the suggestion (although it is not directly the SDK but the API) to search for objects with other keys and not strictly the object name. That's all from me, thanks for the speed in responding to me. Have a good day.
